core/src/consensus/tendermint/engine.rs

```diff
@@ -472,6 +472,12 @@ fn give_additional_rewards<F: FnMut(&Address, u64) -> Result<(), Error>>(
     let sorted_validators = missed_signatures
         .into_iter()
         .map(|(address, (proposed, missed))| (address, Ratio::new(missed, proposed)))
+        // When one sees the Ratio crate's Order trait implementation, he/she can easily realize the
+        // comparing routine is erroneous. It inversely compares denominators when the numerators are same.
+        // That's problematic when it compares such as Ratio{numer: 0, denom:2} and Ratio{numer: 0, denom: 3}.
```

Comment:
// That's problematic when it makes comparisons between ratios such as Ratio{numer: 0, denom:2} and Ratio{numer: 0, denom: 3}.

Reply:
Accepted